src/main-process/atom-application.js

```diff
@@ -255,7 +261,14 @@ class AtomApplication extends EventEmitter {
   async launch (options) {
     if (!this.configFilePromise) {
       this.configFilePromise = this.configFile.watch()
-      this.disposable.add(await this.configFilePromise)
+
+      // TodoElectronIssue: In electron v2 awaiting the watcher causes some delay
+      // in Windows machines, which affects directly the startup time.
+      if (process.platform === 'win32') {
+        this.configFilePromise.then(disposable => this.disposable.add(disposable))
+      } else {
+        this.disposable.add(await this.configFilePromise)
```

Comment:
The `this.config.onDidChange` calls below this `await` really do need to happen *after* the config file is loaded. That is what is causing #19323. I have a fix in the works and will get it out later today.

Reply:
Fixed in #19354.